storage_s3/indico_storage_s3/test.py

```diff
@@ -0,0 +1,71 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import mock
+import pytest
+from freezegun import freeze_time
+
+import indico_storage_s3.plugin as plugin
+from indico_storage_s3.task import create_bucket
+
+
+@pytest.mark.parametrize(('date', 'name_template', 'expected_name'), (
+    ('2018-04-11', 'name', 'name'),
+    ('2018-04-11', 'name-<year>', 'name-2018'),
+    ('2018-04-11', 'name-<year>-<month>', 'name-2018-04'),
```

Comment:
just thinking, should we reject nonsense like month/week without a year?

Reply:
I can add test for that, I think it makes no sense to allow it (and the task currently won't create anything in such cases)

Reply:
I meant throwing an exception when instantiating a storage with such a template